### PR TITLE
[ADD] add module email_prevent_autoreply

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

File: email_restrict_autoreply/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+
+from . import models
```
Comment (alihashaam) [inline email_restrict_autoreply/__init__.py]: CI issue. No new line at the end of the file

File: email_restrict_autoreply/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+
+from . import purchase
```
Comment (alihashaam) [inline email_restrict_autoreply/models/__init__.py]: CI issue. No new line at the end of the file

File: email_restrict_autoreply/models/purchase.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    _inherit = 'mail.message'
+
+    @api.multi
+    def _notify_recipients(self, rdata, record, msg_vals,
+                           force_send=False, send_after_commit=True,
+                           model_description=False, mail_auto_delete=True):
+        channels = rdata.get('channels')
+        partner_obj = self.env['res.partner'].sudo() 
+        IncomingMails = self.env['fetchmail.server'].sudo().search([])
+        partners = [(partner) for partner in rdata.get('partners') if partner_obj.browse(partner.get('id')) and partner_obj.browse(partner.get('id')).email in IncomingMails.mapped('user')]
```
Comment (alihashaam) [inline email_restrict_autoreply/models/purchase.py]: CI issue. Line too long

File: email_restrict_autoreply/models/purchase.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        rdata={'partners': partners, 'channels': channels}
+        return super(MailMessage, self)._notify_recipients(rdata=rdata, record=record, msg_vals=msg_vals,
+                           force_send=force_send, send_after_commit=send_after_commit,
+                           model_description=model_description, mail_auto_delete=mail_auto_delete)
```
Comment (alihashaam) [inline email_restrict_autoreply/models/purchase.py]: CI issue. No new line at the end of the file

File: email_restrict_autoreply/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    "category": "Purchase",
+    "author": "initOS GmbH, Odoo Community Association (OCA)",
+    "website": "http://www.initos.com",
+    "summary": """This module helps to restrict sending mail from external mail replies to all followers of the particular record""",
+    "depends": ["purchase"],
+    "data": [],
+    "license": "AGPL-3",
+    "installable": True,
+    'auto_install': False,
+}
```
Comment (alihashaam) [inline email_restrict_autoreply/__manifest__.py]: CI issue. No new line at the end of the file

**Review by alihashaam — CHANGES_REQUESTED** [review]

**Comment by alihashaam** [comment]
@dsolanki-initos 

Can you please fix the CI issues and please add ticket number in comments as it makes it easy to navigate. 

@jans23 I assigned the PR to myself

**Review by alihashaam — CHANGES_REQUESTED** [review]
Changes requested

**Comment by dsolanki-initos** [comment]
@alihashaam, I've fixed the CI issues as per your feedback.

**Pull request merged**